lib/phoenix_live_view/html_engine.ex

```diff
@@ -880,7 +924,30 @@ defmodule Phoenix.LiveView.HTMLEngine do
       end
 
     {slots, state} = pop_slots(state)
-    attrs = attrs ++ [{:inner_block, [inner_block_assigns]} | slots]
+
+    slot_assigns =
+      if function_exported?(mod, :__components__, 0) do
```

Comment:
We cannot assume the module will be available at this point. The idea in `html_engine` is to store all caller information, without gathering anything from the callee. We should remove this branch. :)